.circleci/config.yml

```diff
@@ -292,15 +302,17 @@ jobs:
           destination: node3/
 
   docker_image_build:
-    <<: *container_config
+    <<: *machine_config
     steps:
       - checkout
-      - *setup_remote_docker
-      - *install_docker_client
       - run:
-          name: Build Docker image
+          name: Build image and smoke test the nodes
           command: |
-            ~/bin/docker build .
+            docker-compose build node1
+            docker-compose up -d
+            sleep 10
+            curl -f -w "\n" localhost:300{1,2,3}/v2/top
```

Comment:
This line seems fragile. I tested stopping the container of node 2 then ran the line locally, and what I got is that a timeout was exercised (slow) and exit status was zero.
```
$ curl -f -w "\n" localhost:300{1,2,3}/v2/top
{"hash":"bh$nyy9XHTWygGem8m9Lj4MUgdaVkchw8o2ee83zJwHPKCS6uHRh","height":11,"nonce":16774641032094533288,"pow":[323,816,939,2281,2845,4324,5058,5447,7510,7721,8223,8380,8505,9824,10110,10846,11816,12488,13823,14816,14874,15076,15188,15790,17304,17932,17958,18141,18258,20466,21346,21544,23255,24952,25326,26389,27089,27348,29507,31486,31766,32720],"prev_hash":"bh$iJeUM1VSGS81F3oUvJjBobvrtYSuLMs5i9RHZzHXCB9drCvjv","state_hash":"bs$2Wc6bhDPKzAiQnLbArYC4yrbqD6xwJMJMGLroyQkDxwVa6Y2iS","target":541777837,"time":1522924696293,"txs_hash":"bx$2DpEMSdcnDud5216VtGfpJ5eJ4de2GzGCEBfMNJ8zGtDZyJ25Q","version":9}

curl: (22) The requested URL returned error: 504 Gateway Time-out
{"hash":"bh$5XKYztdfW5S8Se3hRdwoVgHgXirpshK8eFo32pP3knkgjzYdx","height":15,"nonce":12129856023078514563,"pow":[1701,1840,3078,3389,3398,4270,4950,5267,5360,6900,7445,7579,8614,10310,10515,10610,10819,10978,11959,11970,12050,12085,14158,14820,14827,15241,15468,18932,19020,19385,19788,20866,22761,24122,24266,24428,26487,26597,29156,29651,30880,32461],"prev_hash":"bh$C1cDfEbznUKAmUiego1XcFbN3DGVhpBaqKYC2NeRxR5Xqze6C","state_hash":"bs$2sqriiUVtWWbRU21UDymVfDy5zaX1YvMdV1qn2wD3nfeJCYuDj","target":541936519,"time":1522924745228,"txs_hash":"bx$2GQu4Yhyw1Cmz5LTz3irMgpcPwDRM8nyW2gMQwQEPJgcKBargg","version":9}
```

Reply:
Stopping container 3 (the last one to be contacted) I got non-zero (22) exit code:
```
$ curl -f -w "\n" localhost:300{1,2,3}/v2/top
{"hash":"bh$zRuYNP3uGt8QecEtpZ3pYUXsL9xojRxDhKJwzrMoXLRJtJTff","height":25,"nonce":17545146911512394545,"pow":[312,854,1502,2607,3912,4572,4602,4832,4980,7220,7818,9002,13351,14167,14957,15549,16739,16998,17625,18339,20062,20169,21647,21809,21848,22344,23333,23533,23955,24577,24763,25432,25628,25709,26234,26728,27096,28547,30048,30406,31556,32154],"prev_hash":"bh$2spHEyKw4h1x3FNtqwcMEuzeiFu88p8Y8wYzubZrQobAEAB6fo","state_hash":"bs$2ED9Xw1swELrhdusrZfvC8FbXCXaWmfk5AnDCHeZmAxxXrE8yr","target":543987610,"time":1522924934918,"txs_hash":"bx$2oAC4VrLg3eRpj3VpqDCmoLVkr2NxuoBu5C36QUkhGiA5HTtwB","version":9}

curl: (22) The requested URL returned error: 502 Bad Gateway

curl: (22) The requested URL returned error: 502 Bad Gateway
```